src/jvm/main/org/jetbrains/kotlinx/lincheck/Reporter.kt

```diff
@@ -50,87 +50,194 @@ enum class LoggingLevel {
     INFO, WARN
 }
 
-internal fun <T> printInColumnsCustom(
-        groupedObjects: List<List<T>>,
-        joinColumns: (List<String>) -> String
+internal fun<T> columnsToString(
```

Comment:
Could you please document this code (applies to the whole file)? It is non-trivial and, therefore, only readable with detailed comments.

Reply:
I added doc-strings for `columnsToString` function, `TableLayout` class and its methods. 
Hope its clarifies the code. 

The naming and logic of other methods (e.g. `appendExecutionScenarioWithResults`) remain the same as prior to this PR, and I believe they are pretty self-explanatory.